### ISSUE TITLE
Outdated tests: built-ins/DataView/prototype/set*/range-check-after-value-conversion.js

### ISSUE THREAD

**Issue by bakkot** [tc39/test262#685]
All of the following tests check setting a value in a DataView performs conversion of the value before checking any bounds. But step 4 of 24.2.1.2 has been changed from ToInteger to ToIndex, which checks that its input is within [0, 2^53-1]. 'Infinity' should be replaced with '2**53-1' or similar, removed outright, or possibly moved into the corresponding index-check-before-value-conversion tests.

Invalidated tests:
[1](https://github.com/tc39/test262/tree/master/test/built-ins/DataView/prototype/setFloat32/range-check-after-value-conversion.js), [2](https://github.com/tc39/test262/tree/master/test/built-ins/DataView/prototype/setFloat64/range-check-after-value-conversion.js), [3](https://github.com/tc39/test262/tree/master/test/built-ins/DataView/prototype/setInt16/range-check-after-value-conversion.js), [4](https://github.com/tc39/test262/tree/master/test/built-ins/DataView/prototype/setInt32/range-check-after-value-conversion.js), [5](https://github.com/tc39/test262/tree/master/test/built-ins/DataView/prototype/setInt8/range-check-after-value-conversion.js), [6](https://github.com/tc39/test262/tree/master/test/built-ins/DataView/prototype/setUint16/range-check-after-value-conversion.js), [7](https://github.com/tc39/test262/tree/master/test/built-ins/DataView/prototype/setUint32/range-check-after-value-conversion.js), [8](https://github.com/tc39/test262/tree/master/test/built-ins/DataView/prototype/setUint8/range-check-after-value-conversion.js).

Ping @leobalter, @anba.

**Comment by leobalter** [comment]
> 'Infinity' should be replaced with '2**53-1' or similar

You're right. The assertion using an Infinity index position is not valid.

That position would throw a RangeError before evaluating the poisoned value.

The respective `index-check-before-value-conversion.js` tests should be updated as the files you mentioned can be completely removed.

**Comment by bakkot** [comment]
I'm not sure they should be completely removed. The tests with '100' are valid; those tests check that an implementation hits the poisoned value before checking that 100 is in bounds for the array.

**Comment by leobalter** [comment]
The assertion with `100` is valid, the others are not adding anything to the coverage on that file. There's a file already covering type conversions.

**Comment by woess** [comment]
I believe the `*/detached-buffer-before-outofrange-byteoffset.js` tests are similarly incorrect/outdated; the expected error for index `Infinity` should be `RangeError` (thrown by `ToIndex`) not `TypeError`, afaict.

**Comment by jugglinmike** [comment]
@woess That sounds right to me. A little more concretely, we can take [built-ins/DataView/prototype/setFloat32/detached-buffer-before-outofrange-byteoffset.js](https://github.com/tc39/test262/blob/021d44822c295dae412364d8db125b78dfe0545a/test/built-ins/DataView/prototype/setFloat32/detached-buffer-before-outofrange-byteoffset.js) as an example:

```js
var sample;
var buffer = new ArrayBuffer(12);

sample = new DataView(buffer, 0);

$DETACHBUFFER(buffer);

assert.throws(TypeError, function() {
  sample.setFloat32(Infinity, 0);
}, "Infinity");
```

The buffer may be detached, but ToIndex comes before IsDetachedBuffer and produces a RangeError for the value Infinity, interrupting the algorithm.

>     4. Let getIndex be ? ToIndex(requestIndex).
>     5. Let numberValue be ? ToNumber(value).
>     6. Let isLittleEndian be ToBoolean(isLittleEndian).
>     7. Let buffer be view.[[ViewedArrayBuffer]].
>     8. If IsDetachedBuffer(buffer) is true, throw a TypeError exception.

@leobalter do you agree? If so, could you include the corrections in this patch?

**Comment by leobalter** [comment]
Sure! I just included them on #699

**Comment by evilpie** [comment]
I think you missed built-ins/DataView/prototype/setFloat64/detached-buffer-after-toindex-byteoffset and built-ins/DataView/prototype/setInt16/detached-buffer-after-toindex-byteoffset, both expect TypeError for Infinity.
